module/zfs/dmu_diff.c

```diff
@@ -81,29 +83,16 @@ report_free_dnode_range(struct diffarg *da, uint64_t first, uint64_t last)
 	return (0);
 }
 
-static int
-report_dnode(struct diffarg *da, uint64_t object, dnode_phys_t *dnp)
+static inline boolean_t
+overflow_multiply(uint64_t a, uint64_t b, uint64_t *c)
```

Comment:
Both gcc and clang provide a `__builtin_mul_overflow()` to reliably check for overflows.

https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html
https://clang.llvm.org/docs/LanguageExtensions.html#checked-arithmetic-builtins